lib/galaxy/tool_util/xsd/galaxy.xsd

```diff
@@ -5848,4 +5905,10 @@ is the only supported options.</xs:documentation>
     </xs:restriction>
   </xs:simpleType>
 
+  <xs:simpleType name="MacroImportType">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="[a-zA-Z0-9_\-\.]+.xml"/>
```

Comment:
I've checked the tools-iuc tools, and this would be fine. I think it wouldn't be a big deal if ultimately it causes a small fraction of tools to fail linting. Best practice should be that the macro file is next to the tool, without space in the path, and for eventual plans of distributing tools differently from what we do today this is a big plus.

Reply:
I'm uneasy with this, it doesn't literally represent the spec which even if we require those files sit next to that file, we don't currently require they have a .xml extension. I guess it isn't a huge deal either way though.

Reply:
Many thanks for the review @mvdbeek and @jmchilton!!

I added the ``.xml`` extension restriction because the [vscode extension for the language server relays on the file extension](https://github.com/davelopez/galaxy-language-server/blob/master/client/package.json#L29) to activate itself. Of course, this can be any other file extension, but it felt natural to use .xml also for the macro files.
Anyway, we can fall back to a simple ``xs:string`` without restrictions or adjust the pattern to better fit the spec.